### ISSUE TITLE
Banner Ad is always shown at the top,Also the interstitial Ad is not at all coming in INTEL XDK,HTML cordova application for Android build

### ISSUE THREAD

**Issue by Techjeb** [floatinghotpot/cordova-admob-pro#144]
Hi,

I am using this plugin in Intel XDK HTML cordova application for Android build but am facing 2 issues.

1) Banner Ad is always shown at the top,even after changing position to BOTTOM_CENTER, I have tried with other positions too but same result no change.

2)Interstitial Ad is not at all displayed even after waiting long time .

Following is the code which am using which is enabled for Banner Ad where as for interstitial Ad I have commented corresponding lines.

This is Test Ad display as you can see in code. I am using required ADID from admob too still What is the problem not sure..

 Would you please guide us as we have been struggling for last couple of days with these issues.

<body id="afui" class="af" onload="onLoad()" onresize="onResize()" onunload="onUnload()">
 
        <script>
    function onLoad() {
        if(( /(ipad|iphone|ipod|android|windows phone)/i.test(navigator.userAgent) )) {
            document.addEventListener('deviceready', initApp, false);
        } else {
            initApp();
        }
    }
            function onUnload() {
                //showInterstitial();
            }
	var admobid = {};
	if( /(android)/i.test(navigator.userAgent) ) { 
		admobid = { // for Android
			banner: 'ca-app-pub-xxxxxxxxxxxxxxxxxx',
			interstitial: 'ca-app-pub-xxxxxxxxxxxxxxxx'
		};
	} else if(/(ipod|iphone|ipad)/i.test(navigator.userAgent)) {
		admobid = { // for iOS
			banner: 'ca-app-pub-6869992474017983/4806197152',
			interstitial: 'ca-app-pub-6869992474017983/7563979554'
		};
	} else {
		admobid = { // for Windows Phone
			banner: 'ca-app-pub-6869992474017983/8878394753',
			interstitial: 'ca-app-pub-6869992474017983/1355127956'
		};
	}
        
    function initApp() {
        
        
		if (! AdMob ) { alert( 'admob plugin not ready' ); return; }
		initAd();
        // display the interrestial ad
         alert( 'calling createBanner Method ad method' );
        createBanner();
       // document.addEventListener("backbutton", onBackKeyDown, false); 
        //prepareInterstitial();
       // showInterstitial();
    }
    
    function initAd(){
        var defaultOptions = {
            adSize: 'SMART_BANNER',
            // width: integer, // valid when set adSize 'CUSTOM'
            // height: integer, // valid when set adSize 'CUSTOM'
            overlap:true,
            position: AdMob.AD_POSITION.BOTTOM_CENTER,
            // offsetTopBar: false, // avoid overlapped by status bar, for iOS7+
            bgColor: 'black', // color name, or '#RRGGBB'
            // x: integer,		// valid when set position to 0 / POS_XY
            // y: integer,		// valid when set position to 0 / POS_XY
            isTesting: true, // set to true, to receiving test ad for testing purpose
            // autoShow: true // auto show interstitial ad when loaded, set to false if prepare/show
        };
        AdMob.setOptions( defaultOptions );
        registerAdEvents();
    }
    // optional, in case respond to events or handle error
    function registerAdEvents() {
        // new events, with variable to differentiate: adNetwork, adType, adEvent
        document.addEventListener('onAdFailLoad', function(data){ 
        	alert('error: ' + data.error + 
        			', reason: ' + data.reason + 
        			', adNetwork:' + data.adNetwork + 
        			', adType:' + data.adType + 
        			', adEvent:' + data.adEvent); // adType: 'banner', 'interstitial', etc.
        });
        document.addEventListener('onAdLoaded', function(data){
            //showInterstitial();
            AdMob.showBanner();
       
            
        });
        document.addEventListener('onAdPresent', function(data){
            
             alert('Ad is being shown');
        
        });
        document.addEventListener('onAdLeaveApp', function(data){});
        document.addEventListener('onAdDismiss', function(data){});
    }
    
    function createBanner() {
        AdMob.createBanner({
    adId: admobid.banner,
    position: AdMob.AD_POSITION.BOTTOM_CENTER,
    autoShow: false,
    success: function(){
    },
    error: function(){
        alert('failed to create banner');
        
    }
});
    	
    }
        
    function prepareInterstitial() {
        alert( 'Preparing interestitial ad');
        AdMob.prepareInterstitial({adId:admobid.interstitial, autoShoW:false});
    }
    function showInterstitial() {
        AdMob.isInterstitialReady(function(isready){
    if(isready) AdMob.showInterstitial();
            else {
                alert( 'Interestitial ad is not ready' ); 
            }
});
    }
    function onResize(){
        var s = document.getElementById('sizeinfo');
        s.innerHTML = "web view: " + window.innerWidth + " x " + window.innerHeight;
    }
</script>

**Comment by floatinghotpot** [comment]
You can copy this file to your project, and directly reference to it in your index.html:
https://github.com/floatinghotpot/cordova-admob-pro/blob/master/test/admob_simple.js

**Comment by Techjeb** [comment]
Thanks for your quick response!

I have not got response for my first issue (Banner ad shown always top) would you please guide me regarding this also.

coming to Interstitial ad thanks for correcting but apart from isInterstitialReady API the shared code is same as mine am I right?

**Comment by floatinghotpot** [comment]
There are lots of code in the feature demo, but often it's enough to use it in very simple way, like admob_simple.js

**Comment by Techjeb** [comment]
my issue is even after setting  AdMob.AD_POSITION.BOTTOM_CENTER, ad is displayed at top not displaying in bottom please help us regarding this

**Comment by floatinghotpot** [comment]
Please directly copy the file {plugin}/test/index.html to your project for a testing.

**Comment by sandymacpherson** [comment]
I have the same problem - on Android my banner ad is always at the top despite me specifying it to be BOTTOM_CENTER. I have tried other positions like "LEFT" just to test it and it always shows at the top. For reference, this works fine on iOS.

**Comment by floatinghotpot** [comment]
Please describe your env & os versions, and better paste your code and logs if there is any.

**Comment by sandymacpherson** [comment]
I'm on a Mac (10.10.4) with Cordova 5.1.1 and the Ionic framework 1.6.4

I have found the cause of my problem - I had an event listener for AdLoaded which called AdMob.showBanner() if the banner successfully loaded. However, I wasn't passing any argument to this and so it was defaulting to the default value.

I had assumed that showBanner() would use the position setting specified earlier in createBanner() and this used to work in the older com.google.cordova.admob plugin. Once I upgraded to Cordova 5+ and cordova-admod-pro this problem appeared.

As mentioned earlier, this problem is only on Android.
